File: Clase1/Lab02/20222828_Lab_2 (1)/20222828_Lab_2/QuestionManager.java
import java.util.ArrayList;
import java.util.List;
import java.util.Random;
class QuestionManager {

	private List<Question> questions;
	public QuestionManager(){
		this.questions = new ArrayList<>();
	}

	public void add(Question pregunta){
		questions.add(pregunta);
	}


	public List<Question>  selectRandomN(int n){
		//se crea una copia de la lista questions
		List<Question> copia = new ArrayList<>(questions);
		//numero random
		Random rand = new Random();
		//lista de pregunta seleccionadas
		List<Question> seleccionadas = new ArrayList<>();
		for(int i=0;i<n && copia.size()>0;i++){
			int index = rand.nextInt(copia.size());
			seleccionadas.add(copia.get(index));
			//Elimina esa pregunta de la copia para que no vuelva a salir la misma
			copia.remove(index);
		}

		return seleccionadas;
	}
}

File: Clase6_Lab4/Laboratorio4-Fuentes-C# (1)/Laboratorio4-Fuentes-C#/AssessmentConsoleApp/QuestionManager.cs
using System;
using System.Collections.Generic;

public class QuestionManager
{
    private List<Question> questions = new List<Question>();

    public void Add(Question question)
    {
        this.questions.Add(question);
    }

    public List<Question> SelectRandomN(int n)
    {
        Random rand = new Random();
        List<Question> toReturn = new List<Question>();
        int i = 0;

        while (i < n)
        {
            Question item = questions[rand.Next(questions.Count)];
            if (!toReturn.Contains(item))
            {
                i++;
                toReturn.Add(item);
            }
        }

        return toReturn;
    }
}